cli/src/commands/sign.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::cli_util::CommandHelper;
+use crate::cli_util::RevisionArg;
+use crate::command_error::CommandError;
+use crate::ui::Ui;
+
+/// Cryptographically sign a revision
+#[derive(clap::Args, Clone, Debug)]
+pub struct SignArgs {
+    /// What key to use, depends on the configured signing backend.
+    #[arg()]
+    key: Option<String>,
+    /// What revision to sign
+    #[arg(long, short, default_value = "@")]
+    revision: RevisionArg,
```

Comment:
Wouldn't it make sense to have this be `revisions: Vec<RevisionArg>`?

This way you can do `jj sign -r main..bookmark` to sign a whole branch.

Reply:
That can still be supported with a single argument. The whole `main..bookmark` string gets passed as a single argument from your shell to the `jj` binary. The binary then decides how to interpret it, and the `jj sign` command will presumably want to accept multiple revisions, as most commands do. Commands that want a single revision still accept revsets; they just validate that the revset contains exactly one revision.

It may still be a good idea to allow multiple revision *arguments* here, so the user can do `jj sign -r X -r Y -r Z` to sign those three revsets.

Reply:
Thanks for the clarification c: and yes `jj sign -r X -r Y -r Z` sounds logical